scripts/overwatch.lic

```diff
@@ -41,21 +44,34 @@ module OverWatch
 
   def self.track_hidden_targets(room_id)
     @hidden_targets = room_id
-    echo "Target hid." if @debug
   end
 
-  def self.push_revealed_targets(target_id, target_noun, target_name)
+  def self.push_revealed_targets(target_id, target_noun, target_name, silent_strike = false)
     @hidden_targets = nil
-    echo "Target located." if @debug
-    if GameObj.targets.any? { |npc| npc.id == target_id }
-      echo "NPC already included" if @debug
-    else
+    echo "push_revealed_targets(#{target_id}, #{target_noun}, #{target_name}, #{silent_strike})" if @debug
+
+    if silent_strike
+      echo "Checking for silent strike against #{target_id} #{target_name}." if @debug
+      result = reget(25, /fades into the surroundings\./, /slips into the shadows\./)
+      unless result.length == 0
+        case result[0]
+        when /With a (?:barely audible hiss|sibilant exhalation), <pushBold\/>\w+ <a exist="(\d+)" noun=" ?\w+">[^<]+<\/a><popBold\/> (?:fades into the surroundings|slips into the shadows)\./
+          if target_id == $1
+            echo "Silent Strike Detected. Target #{target_id} #{target_name} is hidden." if @debug
+            return
+          end
+        end
+      end
+    end
+    echo "Silent Strike not detected or failed." if silent_strike && @debug
+
+    unless GameObj.targets.any? { |npc| npc.id == target_id }
       GameObj.new_npc(target_id, target_noun, target_name.gsub(/  /, " "))
+      echo "#{target_id} added to GameObj.npcs."
     end
-    if XMLData.current_target_ids.include?(target_id)
-      echo "ID already in current_target_ids" if @debug
-    else
+    unless XMLData.current_target_ids.include?(target_id)
       XMLData.current_target_ids.unshift(target_id)
+      echo "#{target_id} added to XMLData.current_target_ids."

```

Comment:
Did you mean to leave in the echos without @debug being set?

Reply:
I did not. Thanks